### DISCUSSION TITLE
Separate invisible view flags.

### DISCUSSION THREAD

**Discussion by ZeeMaji** [OpenRCT2/OpenRCT2#22681] [Feature requests]
#22659 changed the behavior of the invisibility toggles in the transparency options window for what i feel is far worse behavior. 

Previously you could have the invisibility flag set regardless of whether or not you had the see through option set for that specific visibility option. Which would let you have things default to being invisible rather than transparent whenever you toggled their visbility in the menu or via hot keys which would help a lot for visibility as i find that the transparency effect isn't great sometimes.

But now this functionality is entirely locked behind the pretty cumbersome transparency options window solely as attempting to enable only the invisibility flag will forcibly also enable the see through flag whenever you reload the park which restricts this functionality solely to the transparency options window.

This is in my opinion greatly inferior to the old behavior where you only had to go to the transparency options window once to toggle what things were invisible when you had them set to see through.

**Comment by tupaschoal** [comment]
This is the chain of events, just for bookkeeping:

1. #16797 introduced new transparency settings (the invisible rather than just the see-through). It also introduced the saving of those invisible flags into the `config.ini` file, but it didn't load and apply them correctly.
2. #17075 started enabling see-through whenever the invisible flag was set via the transparency window.
3. 9e1b24d977efc5a436d331496d383bfdb55c96cf accidentally set to hide supports by default, but no one noticed because loading from `config.ini` was broken. This did make most of the people that played any version from Aug 2022 to 2024 taint their `config.ini` with `invisible_supports = true`
4. #22659 found and fixed the loading of the invisible flags at startup, this included the title screen, which broke the title sequence.
5. #22670 fixed the title screen to not be affected by the invisible settings. It broke the sound configuration
6. #22671 reverted the hide supports by default (3), fixed the sound bug (5) and made untoggling see-through also untoggle invisible as well as untoggling invisible to also untoggle see-through.

So as one can see, most of the recent fixes were trying to fix supposedly wanted behavior from a few years back, but due to how the `config.ini` was tainted, this caught many users offguard.

@AaronVanGeffen suggested that the best way going forward is to make these settings be stored on the save file, rather than the config, and I agree. Opening up for discussion in case anyone sees a drawback with that approach.

PS: review #22679 if this discussion moves faster.

**Reply by MASHzj** [reply]
Will it then be possible again to toggle between visible and invisible with hotkeys, but we'll have to set that for each park instead of once for the whole game (via transparency window)?

**Reply by tupaschoal** [reply]
> Will it then be possible again to toggle between visible and invisible with hotkeys, but we'll have to set that for each park instead of once for the whole game (via transparency window)?

It is still possible to toggle them via hotkeys and I don't see why that would change, but then it would only persist per save
